software/glasgow/device/hardware.py

```diff
@@ -113,6 +112,19 @@ def __init__(self, serial=None, firmware_filename=None, *, _factory_rev=None):
                 # Give every device we loaded firmware onto a bit of time to reenumerate.
                 time.sleep(1.0)
 
+        return handles
+
+    @classmethod
+    def get_serial_list(cls, firmware_filename=None, *, _factory_rev=None):
+        usb_context = usb1.USBContext()
```

Comment:
```suggestion
        with usb1.USBContext() as usb_context:
```
and then you can ditch the explicit `close`. (We can't quite do this in `__init__`, but this is the preferred method when possible because it closes on exception.)

Reply:
I considered a `try / finally` block, but didn't know `USBContext` was a context handler (I guess it's in the name!), thanks.